service/history/shard/context_impl.go

```diff
@@ -1933,7 +1933,18 @@ func (s *ContextImpl) acquireShard() {
 		return nil
 	}
 
-	err := backoff.ThrottleRetry(op, policy, common.IsPersistenceTransientError)
+	// keep retrying except ShardOwnershipLostError or lifecycle context ended
+	acquireShardRetryable := func(err error) bool {
+		if s.lifecycleCtx.Err() != nil {
```

Comment:
you don't really have to check `lifecycleCtx` here: the first thing `op` does is check `isValid`, which checks if state >= stopping, and if so returns a shardownershiplost error. state >= stopping iff lifecycleCtx is cancelled.

I'd prefer not doing this additional check since it makes the code more confusing (to me). but it doesn't hurt